app/uk/gov/hmrc/cdsreimbursementclaimfrontend/controllers/claims/CheckDuplicateDeclarationDetailsController.scala

```diff
@@ -53,11 +54,10 @@ class CheckDuplicateDeclarationDetailsController @Inject() (
     authenticatedActionWithSessionData.async { implicit request =>
       val isDuplicate: Boolean = true
       withAnswersAndRoutes[DisplayDeclaration] { (_, maybeDeclaration, router) =>
+        implicit val reimbursementRoutes: ReimbursementRoutes = router
```

Comment:
this is nasty - we will need to look at this later

Reply:
Yes, perhaps change withAnswersAndRoutes implementation? 

The problem is that to sufficiently render an answer which consist of multiple fields some keys must be mixed with sub keys as this for example
```
      SummaryListRow(
        key = Key(Text(messages(lang(key, router.subKey, "mrn-label")))),
        value = Value(Text(declaration.displayResponseDetail.declarationId))
      ).some,
```
and some not 
```
      SummaryListRow(
        key = Key(Text(messages(s"$key.import-date-label"))),
        value = Value(Text(declaration.displayResponseDetail.acceptanceDate))
      ).some,
```
Also answers  must access journey bindable.

It appears that only way to fully satisfy this requirements is to bring Router which that contains that necessary information.

Besides `Router` is really good candidate to carry it over as implicit. It's same dependency as messages, helper utility, which carried implicitly.

Reply:
`withAnswersAndRoutes` is the problem here and the Router design in general - not a problem to solve in this PR though - we can review this later.

Reply:
ok